app/controllers/Application.scala

```diff
@@ -96,16 +96,18 @@ class Application(
     )).withSettingsSurrogateKey
   }
 
-  def contributionsLanding(countryCode: String): Action[AnyContent] = CachedAction() { implicit request =>
+  def oldOrNewContributionsLanding(countryCode: String): Action[AnyContent] = maybeAuthenticatedAction().async { implicit request =>
+    type Attempt[A] = EitherT[Future, String, A]
     implicit val settings: Settings = settingsProvider.settings()
-    Ok(views.html.main(
-      title = "Support the Guardian | Make a Contribution",
-      description = stringsConfig.contributionsLandingDescription,
-      mainId = s"contributions-landing-page-$countryCode",
-      mainJsBundle = "contributionsLandingPage.js",
-      mainStyleBundle = "contributionsLandingPageStyles.css",
-      scripts = views.html.addToWindow("paymentApiPayPalEndpoint", paymentAPIService.payPalCreatePaymentEndpoint)
-    )).withSettingsSurrogateKey
+    val experiments = settings.switches.experiments
+    if (experiments.get("newPaymentFlow").exists(_.isParticipating)) {
```

Comment:
this settings.switches thing, how does it have an isParticipating thing, surely participation is a user level thing rather than a switch?

Reply:
It's `ExperimentSwitch` that has an `isParticipating` flag. I guess they're under switches because they have an on/off property like other non-A/B-test switches. Maybe that's confusing but it's following the model of dotcom where A/B tests are on the switchboard along with features.